### ISSUE TITLE
prefer-for-of rule recommended on a node list

### ISSUE THREAD

**Issue by mattferderer** [palantir/tslint#2927]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: VSCode eg2.tslint

#### TypeScript code being linted

```ts
const items = document.getElementsByClassName("items")
for(let i =0; i < items.length; i++) {
  
}
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {
      "semicolon": [true, "never"],
      "no-unnecessary-type-assertion": true
    },
    "rulesDirectory": []
}
```

#### Actual behavior
Receiving the error "Expected a 'for-of' loop instead of a 'for' loop with this simple iteration (prefer-for-of)
#### Expected behavior
A for of loop to my knowledge shouldn't be used on a NodeList as it's not supported widely & is also not part of any standard. TypeScript will complain if you do this. The same with a forEach loop.

**Comment by Kovensky** [comment]
It is specified as `iterable<Node>` (https://dom.spec.whatwg.org/#interface-nodelist). If TypeScript does not allow you to iterate on it, then it's TypeScript that needs fixing.

Also note that core-js can polyfill iterators for most DOM collections, including NodeList.

If you need to be "safe" and avoid iterating on them directly, you could also wrap the `NodeList`-resulting call with `Array.from`.

**Comment by mattferderer** [comment]
@Kovensky do you run a specific tsconfig that allows TypeScript to do this or are you suggesting to request it as an issue for them?

**Comment by NN---** [comment]
@Kovensky It is iterable with ES6 support or with --downlevelIterators flag in TypeScript.
If you use ES5 mode , then it is not iterable using for..of.

I had to remove this rule due to this false positive.
